masonry/src/widget/text_area.rs

```diff
@@ -477,6 +477,32 @@ impl<const EDITABLE: bool> TextArea<EDITABLE> {
         // Determine whether there's any advantage to that
         this.ctx.request_layout();
     }
+
+    /// Set the selection to the given byte range.
+    ///
+    /// No-op if either index is not a char boundary.
+    pub fn select_byte_range(this: &mut WidgetMut<'_, Self>, start: usize, end: usize) {
+        let (fctx, lctx) = this.ctx.text_contexts();
+        this.widget
+            .editor
+            .driver(fctx, lctx)
+            .select_byte_range(start, end);
+        this.ctx.request_render();
```

Comment:
Why is this needed? You request a layout directly below

Reply:
Oh wow, I left a lot of cruft from debugging.